packages/next/client/index.js

```diff
@@ -240,13 +240,54 @@ export async function renderError (props) {
 // If hydrate does not exist, eg in preact.
 let isInitialRender = typeof ReactDOM.hydrate === 'function'
 function renderReactElement (reactEl, domEl) {
+  // mark start of hydrate/render
+  if (typeof performance !== 'undefined') {
```

Comment:
I think we may need to safeguard these further in order to account for browsers like IE11 that have `performance.now` but not `performance.mark`, currently they'd throw. Thoughts?

Reply:
Good call, I added some constants to `next-server/utils` that does this

(if someone thinks theres a better place to put these constants, please let me know)